RevenueCatUI/Templates/Components/TemplateComponentsView.swift

```diff
@@ -0,0 +1,202 @@
+//
+//  File.swift
+//  
+//
+//  Created by Josh Holtz on 6/11/24.
+//
+
+import RevenueCat
+import SwiftUI
+// swiftlint:disable all
+
+#if PAYWALL_COMPONENTS
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+private class ComponentPaywallData: ObservableObject {
+    @Published var selectedPackage: TemplateViewConfiguration.Package
+
+    init(selectedPackage: TemplateViewConfiguration.Package) {
+        self._selectedPackage = .init(initialValue: selectedPackage)
+    }
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct TemplateComponentsView: TemplateViewType {
+
+
+    @StateObject
+    private var componentPaywallData: ComponentPaywallData
+
+    let configuration: TemplateViewConfiguration
+
+    @Environment(\.userInterfaceIdiom)
+    var userInterfaceIdiom
+    @Environment(\.locale)
+    var locale
+
+    #if swift(>=5.9) || (!os(macOS) && !os(watchOS) && !os(tvOS))
+    @Environment(\.verticalSizeClass)
+    var verticalSizeClass
+    #endif
+
+    init(_ configuration: TemplateViewConfiguration) {
+        self.configuration = configuration
+        self._componentPaywallData = .init(wrappedValue: .init(selectedPackage: configuration.packages.default))
+    }
+
+    var body: some View {
+        VStack(spacing: 0) {
+            if let data = self.configuration.components {
+                ComponentsView(
+                    locale: self.locale,
+                    components: data.components,
+                    configuration: self.configuration,
+                    shouldSplitLandscape: true
+                )
+                .environmentObject(self.componentPaywallData)
+            }
+        }
+        .edgesIgnoringSafeArea(.top)
+        .background(
+            try! PaywallColor(stringRepresentation: self.configuration.components!.backgroundColor.light).underlyingColor
+        )
+    }
+
+}
+
+func getLocalization(_ locale: Locale, _ displayString: DisplayString) -> String {
+    if let found = displayString.value[locale.identifier] {
+        return found
+    }
+
+    return displayString.value.values.first!
```

Comment:
same question here, what does it mean if we just couldn't find anything at the end of this chain?

Reply:
This will be cleaned up.